crates/matrix-sdk/src/client/mod.rs

```diff
@@ -124,11 +134,16 @@ pub enum LoopCtrl {
     Break,
 }
 
-/// Wrapper struct for ErrorKind::UnknownToken
+/// Represents changes that can occur to a `Client`s `Session`.
 #[derive(Debug, Clone)]
-pub struct UnknownToken {
-    /// Whether or not the session was soft logged out
-    pub soft_logout: bool,
+pub enum SessionChange {
+    /// The session's token is no longer valid.
+    UnknownToken {
+        /// Whether or not the session was soft logged out
+        soft_logout: bool,
+    },
+    /// The session's tokens have been refreshed.
+    TokensRefreshed,
```

Comment:
You know it is possible to listen to tokens change by subscribing to the auth API's `session_tokens_stream`?

Reply:
I did not, no. I've played with these however, and the streams are always `None` in the FFI's `Client::new` so I'm going to stick with this solution as it works.